### ISSUE TITLE
Stats overhaul breaks date stat results and BC because of return type adjustments

### ISSUE THREAD

**Issue by tscni** [solariumphp/solarium#957]
**Solarium version(s) affected**: 6.1.5

**Description**  
The stats overhaul PR (https://github.com/solariumphp/solarium/pull/943) introduced an issue with `Result\Stats\Result`.
The return type of `::getMin()` and `::getMax()` was previously `?string` and is now `?float`.

The float return only works for numeric field types, but other types like dates obviously don't return floats.
The methods in `Result\Stats\FacetValue` on the other hand already returned `?float`, which I guess was actually incorrect in that regard.

Either way, merging the functions between `Result` and `FacetValue` into `ResultTrait` is a BC break because of the type adjustments.

See:
- https://github.com/solariumphp/solarium/blob/6.1.5/src/Component/Result/Stats/ResultTrait.php#L29
- https://github.com/solariumphp/solarium/blob/6.1.4/src/Component/Result/Stats/Result.php#L58
- https://github.com/solariumphp/solarium/blob/6.1.4/src/Component/Result/Stats/FacetValue.php#L58

**Comment by thomascorthals** [comment]
You are absolutely right! I did the overhaul based on the ref guide and they only provide [examples](https://solr.apache.org/guide/8_9/the-stats-component.html#stats-component-example) with floats.

```xml
    <lst name="price">
      <double name="min">0.0</double>
      <double name="max">2199.0</double>
      <long name="count">16</long>
      <long name="missing">16</long>
      <double name="sum">5251.270030975342</double>
      <double name="sumOfSquares">6038619.175900028</double>
      <double name="mean">328.20437693595886</double>
      <double name="stddev">536.3536996709846</double>
    </lst>
```

`::getMean()` is also affected.

```xml
    <lst name="manufacturedate_dt">
      <date name="min">2005-08-01T16:30:25Z</date>
      <date name="max">2006-02-14T23:55:59Z</date>
      <long name="count">11</long>
      <long name="missing">22</long>
      <double name="sum">1.2510623166E13</double>
      <date name="mean">2006-01-15T12:49:38.727Z</date>
      <double name="sumOfSquares">1.4229031714649898E25</double>
      <double name="stddev">5.765776375213974E9</double>
    </lst>
```

On the plus side, only one place to fix now that it's in a trait.

**Comment by thomascorthals** [comment]
This was fixed in [Solarium 6.1.6](https://github.com/solariumphp/solarium/releases/tag/6.1.6).
